tiledb/common/memory_tracker.cc

```diff
@@ -147,8 +147,12 @@ uint64_t MemoryTrackerResource::get_count() {
 }
 
 void* MemoryTrackerResource::do_allocate(size_t bytes, size_t alignment) {
-  total_counter_.fetch_add(bytes, std::memory_order_relaxed);
+  auto previous_total =
+      total_counter_.fetch_add(bytes, std::memory_order_relaxed);
```

Comment:
On second look using `memory_order_relaxed` makes the trackers not thread-safe. 🤔